### PR TITLE
#505: create startup script command

### PR DESCRIPTION
**Pull request by jan-vcapgemini**
Fixes: #440, #366, #505

Implements:
* added startup script command for intellij and android-studio for Windows, Linux and MacOS
* removed getBinaryName

There is one issue left. On Mac starting `android-studio` does not work because `studio` does not have execute permissions.

### CODE REVIEW

File: cli/src/main/java/com/devonfw/tools/ide/tool/intellij/Intellij.java [hunk 1]
```diff
@@ -68,6 +63,56 @@ protected void postInstall() {
...
+    if (this.context.getSystemInfo().isMac()) {
+      try {
+        bashFile = Files.createFile(getToolBinPath().resolve(getName()));
+        Files.writeString(bashFile,
+            bashFileContentStart + getToolPath().resolve("IntelliJ IDEA CE.app").resolve("Contents").resolve("MacOS").resolve(IDEA) + bashFileContentEnd);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+      // Setting execute permissions is only required if executed on a real MacOS, won't work on Windows.
+      if (SystemInfoImpl.INSTANCE.isMac()) {
```
Comment (hohwille) [inline cli/src/main/java/com/devonfw/tools/ide/tool/intellij/Intellij.java:83]: Must also be done on Linux.
Why is the method called `setMacOsFilePermissions`.
I commented about this topic in several PRs already.
IMHO we need a `setExecutable` or `makeExecutable` method in `FileUtil` that does `chmod a+x` on a given file and will do nothing on Windows (and not fail there).

File: cli/src/main/java/com/devonfw/tools/ide/tool/intellij/Intellij.java [hunk 2]
```diff
@@ -68,6 +63,56 @@ protected void postInstall() {
...
+  private void createStartupCommandScript() {
+    this.context.getFileAccess().mkdirs(getToolPath().resolve("bin"));
+    Path bashFile;
+    String bashFileContentStart = "#!/usr/bin/env bash\n'";
+    String bashFileContentEnd = "' $*";
+    if (this.context.getSystemInfo().isMac()) {
+      try {
+        bashFile = Files.createFile(getToolBinPath().resolve(getName()));
+        Files.writeString(bashFile,
+            bashFileContentStart + getToolPath().resolve("IntelliJ IDEA CE.app").resolve("Contents").resolve("MacOS").resolve(IDEA) + bashFileContentEnd);
```
Comment (hohwille) [inline cli/src/main/java/com/devonfw/tools/ide/tool/intellij/Intellij.java:78]: Please check how this was done in `devonfw-ide`.
It should go to `IdeToolCommandlet` and the script should simply use IDEasy itself to launch the IDE.
Any other approach is cursed to fail from the start.

File: cli/src/main/java/com/devonfw/tools/ide/tool/intellij/Intellij.java (hunk 1 above)
Comment (hohwille) [reply cli/src/main/java/com/devonfw/tools/ide/tool/intellij/Intellij.java:83]: Created issue #542

Comment (jan-vcapgemini) [reply cli/src/main/java/com/devonfw/tools/ide/tool/intellij/Intellij.java]: I've implemented the changes of #542 here too now.

Comment (hohwille) [reply cli/src/main/java/com/devonfw/tools/ide/tool/intellij/Intellij.java]: Excellent solution. 👍

**Review by hohwille — APPROVED** [review]
@jan-vcapgemini nicely completed. Great PR and finally solved various problems and previous workarounds. 👍

**Pull request merged**